### DISCUSSION TITLE
Process imported rows before saving

### DISCUSSION THREAD

**Discussion by w0r00d** [filamentphp/filament#15198] [Help] [answered]
### Package

Actions

### Package Version

v3.0.0

### How can we help you?

I am using import action to import CSV file into a resource. Is there a way to process rows together before storing them?

By processing I mean executing some queries on the imported rows and continue with the saving or not based on conditions.

**Comment by leandrocfe** [comment, marked as answer]
Did you check [lifecycle hooks](https://filamentphp.com/docs/3.x/actions/prebuilt-actions/import#lifecycle-hooks)?

**Reply by w0r00d** [reply]
Lifecycle hooks are useful yes but they handle the imported data row by row. To implement my logic it seems that I need to implement custom import logic with livewire.
